launch_ros/launch_ros/actions/node.py

```diff
@@ -419,6 +419,30 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
         self.__final_node_name = prefix_namespace(
             self.__expanded_node_namespace, self.__expanded_node_name)
 
+        # Check to see if a global parameter file was provided
+        # First extract the raw name of the node
+        node_name_as_text = None
+        if type(self.__node_name) == list:
+            node_name_as_text = list(self.__node_name[0].__dict__.values())[0]
+        elif type(self.__node_name) == str:
+            node_name_as_text = self.__node_name
+
+        params_from_file = context.launch_configurations.get('global_params_from_file', None)
+        node_params_from_file = None
+        # Check if ROS parameters were provided in the file for this node
+        if params_from_file is not None and node_name_as_text is not None:
+            if node_name_as_text in params_from_file:
+                if 'ros__parameters' in params_from_file[node_name_as_text]:
+                    node_params_from_file = params_from_file[node_name_as_text]['ros__parameters']
```

Comment:
you can already use wildcards in node names, e.g. /**.
Maybe it's easier to pass the parameter file to all nodes, `rcl` will already only read the parameters that actually applied to the node

Reply:
I added this in case we wanted to pass the same parameter with different values for different nodes. For e.g., ```demo_node_1``` has ```param_2: 5``` and ```demo_node_2``` has ```param_2: 10``` in the example above.